src/compiler/checker.ts

```diff
@@ -6814,37 +6814,58 @@ namespace ts {
             return undefined;
         }
 
-        function resolveTypeReferenceName(typeReferenceName: EntityNameExpression | EntityName) {
+        function resolveTypeReferenceName(node: TypeReferenceType, typeReferenceName: EntityNameExpression | EntityName) {

```

Comment:
This function is only called in one location. At that location, we have special handling for JSDocTypeReference already. So maybe we should just avoid calling `resolveTypeReferenceName` from that code path altogether instead of special-casing it here. Same for `getTypeReferenceName`.